src/mkdocstrings/handlers/__init__.py

```diff
@@ -0,0 +1,242 @@
+
+
+def do_any(seq: Sequence, attribute: str = None) -> bool:
+    """
+    The `any` builtin as a filter for Jinja templates.
+
+    Args:
+        seq: An iterable object.
+        attribute: The attribute name to use on each object of the iterable.
+
+    Returns:
+        A boolean telling if any object of the iterable evaluated to True.
+    """
+    if attribute is None:
+        return any(seq)
+    return any(o[attribute] for o in seq)
+
+
+class BaseRenderer:
+    """
+    The base renderer class.
+
+    Inherit from this class to implement a renderer.
+
+    You will have to implement the `render` method.
+    You can also override the `update_env` method, to add more filters to the Jinja environment,
+    making them available in your Jinja templates.
+
+    To define a fallback theme, add a `FALLBACK_THEME` class-variable.
+    """
+
+    def __init__(self, directory: str, theme: str) -> None:
+        """
+        Initialization method.
+
+        If the given theme is not supported (it does not exist), it will look for a `FALLBACK_THEME` attribute
+        in `self` to use as a fallback theme.
+
+        Arguments:
+            directory: The name of the directory containing the themes for this renderer.
+            theme: The name of theme to use.
+        """
+        themes_dir = Path(__file__).parent.parent / "templates" / directory
+        theme_dir = themes_dir / theme
+        if not theme_dir.exists():
+            if hasattr(self, "FALLBACK_THEME"):
+                log.warning(
+                    f"mkdocstrings.handlers: No '{theme}' theme in '{directory}', "
+                    f"falling back to theme '{self.FALLBACK_THEME}'"
+                )
+                theme_dir = themes_dir / self.FALLBACK_THEME
+            else:
+                raise ThemeNotSupported(theme)
+
+        self.env = Environment(autoescape=True, loader=FileSystemLoader(theme_dir))
+        self.env.filters["highlight"] = do_highlight
+        self.env.filters["any"] = do_any
+
+    def render(self, data: DataType, config: dict) -> str:
+        """
+        Render a template using provided data and configuration options.
+
+        Arguments:
+            data: The collected data to render.
+            config: The rendering options.
+
+        Returns:
+            The renderer template as HTML.
+        """
+        raise NotImplementedError
+
+    def update_env(self, md: Markdown, config: dict) -> None:
+        """
+        Update the Jinja environment.
+
+        Arguments:
+            md: The Markdown instance. Useful to add functions able to convert Markdown into the environment filters.
+            config: Configuration options for `mkdocs` and `mkdocstrings`, read from `mkdocs.yml`. See the source code
+              of [mkdocstrings.plugin.MkdocstringsPlugin.on_config][] to see what's in this dictionary.
+        """
+        md = Markdown(extensions=config["mdx"], extensions_configs=config["mdx_configs"])
```

Comment:
I'm not sure what's going on here, but `Markdown` doesn't accept an `extensions_configs` parameter; it gets silently ignored. `extension_configs` is a thing, though.

https://github.com/Python-Markdown/markdown/blob/5891418956d08ea3ddcbf65281e6882c7e3d9662/markdown/core.py#L96

Reply:
It definitely looks like a typo, and that should really be fixed!